src/consoleappender.cxx

```diff
@@ -103,7 +131,14 @@ ConsoleAppender::append(const spi::InternalLoggingEvent& event)
     thread::MutexGuard guard (getOutputMutex ());
 
     tostream& output = (logToStdErr ? tcerr : tcout);
+
+	auto cur_loc = output.getloc();
```

Comment:
I think we want to make this conditional, only do this if user specifies different than default locale. Also, we should call flush before changing the locale and flush after we change the locale.